### PR TITLE
[doc] update gradient accumulation

### PR DESCRIPTION
**Pull request by jiangmingyan**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

#3769

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

update update gradient accumulation document.

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: docs/source/en/features/gradient_accumulation.md [hunk 1]
```diff
@@ -18,20 +15,145 @@ and only update the parameters in the preset iteration.
...
+### Step 2. Import libraries in train.py
+Create a `train.py` and import the necessary dependencies. The version of `torch` should not be lower than 1.8.1.
+
+```python
+import os
+from contextlib import nullcontext
+from pathlib import Path
+
+import torch
+from titans.utils import barrier_context
```
Comment (ver217) [inline docs/source/en/features/gradient_accumulation.md:47]: You can use https://github.com/hpcaitech/ColossalAI/blob/73d3e4d3091a6f87cd5153194989713dd2429210/colossalai/cluster/dist_coordinator.py#L125

File: docs/sidebars.json [hunk 2]
```diff
@@ -45,6 +45,7 @@
         "features/gradient_clipping",
         "features/gradient_handler",
         "features/zero_with_chunk",
+        "features/gradient_accumulation_with_booster",
```
Comment (ver217) [inline docs/sidebars.json:48]: Move this above the old grad accumulation doc

File: docs/source/en/features/gradient_accumulation_with_booster.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+- [Define Your Configuration](../basics/define_your_config.md)
+- [Training Booster](../basics/booster_api.md)
+
+## Introduction
+
+Gradient accumulation is a common way to enlarge your batch size for training. When training large-scale models, memory can easily become the bottleneck and the batch size can be very small, (e.g. 2), leading to unsatisfactory convergence. Gradient accumulation works by adding up the gradients calculated in multiple iterations, and only update the parameters in the preset iteration.
+
+## Usage
+
+It is simple to use gradient accumulation in Colossal-AI. Just call `booster.no_sync()` which returns a context manager. It accumulate gradients without synchronization, meanwhile you should not update the gradients.
```
Comment (ver217) [inline docs/source/en/features/gradient_accumulation_with_booster.md:15]: > meanwhile you should not update the gradients.

I guess it may be "meanwhile you should not update the weights". Right?

File: docs/source/en/features/gradient_accumulation_with_booster.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+args = parser.parse_args()
+# launch from torch
+colossalai.launch_from_torch(config=dict())
+```
+
+### Step 3. Create training components
+Build your model, optimizer, loss function, lr scheduler and dataloaders. Note that the root path of the dataset is obtained from the environment variable `DATA`. You may `export DATA=/path/to/data` or change `Path(os.environ['DATA'])` to a path on your machine. Data will be automatically downloaded to the root path.
+
+```python
+    # define the training hyperparameters
```
Comment (ver217) [inline docs/source/en/features/gradient_accumulation_with_booster.md:56]: Why do all lines start with a tab?

File: docs/source/en/features/gradient_accumulation_with_booster.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+            with sync_context:
+                output = model(img)
+                train_loss = criterion(output, label)
+                booster.backward(train_loss, optimizer)
+        else:
+            output = model(img)
+            train_loss = criterion(output, label)
+            booster.backward(train_loss, optimizer)
+            optimizer.step()
+            model.zero_grad()
```
Comment (ver217) [inline docs/source/en/features/gradient_accumulation_with_booster.md:112]: `optimizer.zero_grad()` may be more common-used.

File: docs/source/zh-Hans/features/gradient_accumulation_with_booster.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+- [定义配置文件](../basics/define_your_config.md)
+- [训练中使用Booster](../basics/booster_api.md)
+
+## 引言
+
+梯度累积是一种常见的增大训练 batch size 的方式。 在训练大模型时，内存经常会成为瓶颈，并且 batch size 通常会很小（如2），这导致收敛性无法保证。梯度累积将多次迭代的梯度累加，并仅在达到预设迭代次数时更新参数。
+
+## 使用
+
+在 Colossal-AI 中使用梯度累积非常简单，booster提供no_sync返回一个文件管理器，在该文件管理器下取消同步并且累积梯度。
```
Comment (ver217) [inline docs/source/zh-Hans/features/gradient_accumulation_with_booster.md:15]: 上下文管理器？

**Review by ver217 — APPROVED** [review]
